src/flux/roster.coffee

```diff
@@ -24,18 +24,35 @@ RosterConfig = {
     @emitChange()
 
   deleted: (unused, studentId) ->
-    # remove the student from all the courses rosters
+    # set inactive
     for courseId, roster of @_local
       students = roster.students
-      index = _.findIndex(students, id: studentId)
-      students.splice(index, 1) unless -1 is index
+      student = _.findWhere(students, id: studentId)
+      student.is_active = false
+    @emitChange()
+
+  undrop: (unused, studentId) ->
+
+  undropped: (unused, studentId) ->
+    # set active
+    for courseId, roster of @_local
+      students = roster.students
+      student = _.findWhere(students, id: studentId)
+      student.is_active = true
     @emitChange()
```

Comment:
You're looping over each course and looking for a student by id, and then setting `is_active` on it without checking if it was found.  If multiple courses are loaded and the student only belongs to one, then the `findWhere` will return null.  Ideally

I think you should pass a courseId in and only search that course.  It'll be surprising behavior for a teacher if they've dropped a student from multiple courses and then attempt to re-add him to one course and it does it to all courses.

The student may still be not found though, it'd probably be good defensive programming to do something like: 
`student?.is_active = true`